### PR TITLE
[T4A3][F12-A2] Aditya Hendraatmaja

### PR DESCRIPTION
**Pull request by ahendraatmaja**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,54 +6,18 @@
...
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
+    	super(address, isPrivate, ADDRESS_VALIDATION_REGEX, MESSAGE_ADDRESS_CONSTRAINTS);
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java:21]: Good job moving validation into `Contact` class =).

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -6,54 +6,18 @@
...
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java:45]: What happens if you do not override `equals(...)` in `Address` class?
Is `Address` equals to `Email`?
If it is not how will you be able to differentiate between the classes?

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact in the address book. Contains 3 sub-classes: address, email, and phone.
+ * Guarantees: immutable; is valid as declared in {@link #isValidContact(String)}
+ */
+public class Contact {
+	
+	protected final String value;
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Contact.java:11]: Do note on indentation.

**Review by brandonyeoxg — COMMENTED** [review]
@ahendraatmaja Some comments added. Please close the PR after reading comments.

File: src/seedu/addressbook/data/person/Address.java (hunk 2 above)
Comment (ahendraatmaja) [reply src/seedu/addressbook/data/person/Address.java:45]: Yup, I only realised this after the discussion in the tutorial.

**Pull request closed without merging**